src/index.js

```diff
@@ -105,8 +106,8 @@ chrome.runtime.onMessage.addListener(async (payload, sender, sendResponse) => {
     id,
     method,
     params,
-    __frameOrigin: getOrigin((tab || {}).url || sender.url),
-    __extensionConnecting: payload.__extensionConnecting
+    __frameOrigin: getOrigin(tab, sender),
+    __extensionConnecting: method === 'embedded_action_res' ? true : payload.__extensionConnecting
```

Comment:
might there ever a time where we do want a request resulting from an embedded action to be tracked by Frame? @floating might know the answer to this question but I think we can limit the scope of these changes to just setting the last parameter to `true` for the embedded `getChainId` call above, since this is the specific call that attempts to get the chain ID on connection:

`getChainId: async () => ({ chainId: await window.ethereum._send('eth_chainId', [], undefined, true) })`

Reply:
I already tried this and it doesn't work - origin added on opening extension window.